schemes/rrtmgp/rrtmgp_sw_calculate_fluxes.F90

```diff
@@ -0,0 +1,109 @@
+      type(ty_fluxes_broadband_ccpp), intent(in) :: fswc                ! Shortwave clear-sky flux object
+      ! Output variables
+      real(kind_phys),               intent(out) :: fns(:,:)            ! Shortwave net radiative flux [W m-2]
+      real(kind_phys),               intent(out) :: fcns(:,:)           ! Shortwave net radiative clear-sky flux [W m-2]
+      real(kind_phys),               intent(out) :: fsns(:)             ! Shortwave net upward flux at surface [W m-2]
+      real(kind_phys),               intent(out) :: fsnt(:)             ! Shortwave net outgoing flux at model top [W m-2]
+      real(kind_phys),               intent(out) :: soll(:)             ! Direct solar radiative flux at surface >= 700nm [W m-2]
+      real(kind_phys),               intent(out) :: sols(:)             ! Direct solar radiative flux at surface < 700nm [W m-2]
+      real(kind_phys),               intent(out) :: solld(:)            ! Diffuse solar radiative flux at surface >= 700nm [W m-2]
+      real(kind_phys),               intent(out) :: solsd(:)            ! Diffuse solar radiative flux at surface < 700nm [W m-2]
+
+      
+      ! CCPP error handling variables
+      character(len=512), intent(out) :: errmsg
+      integer,            intent(out) :: errflg
+
+      ! Local variables
+      integer :: diag_index, idx
+      real(kind_phys), dimension(size(fsw%fluxes%bnd_flux_dn,1), &
+                                 size(fsw%fluxes%bnd_flux_dn,2), &
+                                 size(fsw%fluxes%bnd_flux_dn,3)) :: flux_dn_diffuse
+
+      errmsg = ''
+      errflg = 0
+
+      diag_index = num_diag_subcycles - icall + 1
+
+      ! Don't do anything if this subcycle is inactive
+      if (.not. active_calls(diag_index)) then
+         return
+      end if
+
+      ! Initialize to provide 0.0 values for night columns.
+      fns = 0.0_kind_phys
+      fcns = 0.0_kind_phys
+      fsns = 0.0_kind_phys
+      fsnt = 0.0_kind_phys
+      soll = 0.0_kind_phys
+      sols = 0.0_kind_phys
+      solld = 0.0_kind_phys
+      solsd = 0.0_kind_phys
+
+      do idx = 1, nday
+         fns(idxday(idx), ktopcam:) = fsw%fluxes%flux_net(idx, ktoprad:)
+         fcns(idxday(idx), ktopcam:) = fswc%fluxes%flux_net(idx, ktoprad:)
+      end do
+
+      fsns(:) = fns(:, pverp)
+      fsnt(:) = fns(:, ktopcam)
+
+      ! Export surface fluxes
+      ! sols(pcols)      Direct solar rad on surface (< 0.7)
+      ! soll(pcols)      Direct solar rad on surface (>= 0.7)
+      ! RRTMGP: Near-IR bands (1-10), 820-16000 cm-1, 0.625-12.195 microns
+      ! Put half of band 10 in each of the UV/visible and near-IR values,
+      ! since this band straddles 0.7 microns:
+      ! UV/visible bands 10-13, 16000-50000 cm-1, 0.200-0.625 micron
+      ! Calculate diffuse flux from total and direct
+      flux_dn_diffuse = fsw%fluxes%bnd_flux_dn - fsw%fluxes%bnd_flux_dn_dir
+
+      do idx = 1, nday
```

Comment:
I might add a comment here noting that these `sol` calculations appear to essentially be hardcoded, and so will require modifications if the RRTMGP flux bands type/array ever changes.